### PR TITLE
Add base proguard rules

### PR DESCRIPTION
**Pull request by luismfonseca**
This PR adds the base default rules for allowing minification using Proguard in Android Studio.

### CODE REVIEW

File: sample_kotlin/build.gradle [hunk 1]
```diff
@@ -15,6 +15,10 @@ android {
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
     }
     buildTypes {
+        debug {
```
Comment (zmarkan) [inline sample_kotlin/build.gradle]: you might want to remove this from the `debug` build type

File: sample_kotlin/build.gradle [hunk 2]
```diff
@@ -15,6 +15,10 @@ android {
...
+            minifyEnabled false
```
Comment (zmarkan) [inline sample_kotlin/build.gradle]: You don't need the minify bit in debug, maybe turn it on in release?

**Comment by zmarkan** [comment]
Just left one comment - otherwise LGTM

File: sample_kotlin/build.gradle (hunk 2 above)
Comment (luismfonseca) [reply sample_kotlin/build.gradle]: I turned this on when testing locally: it was easier to test the proguard this way, without going through the hassle of doing a release build.

**Pull request merged**